### PR TITLE
Tiny improvements to systemd docs.

### LINKED ISSUE #13796: Tiny improvements to systemd docs
I feel like the [systemd article](http://docs.master.dockerproject.org/articles/systemd/) could be improved just a tiny bit by:
* showing how to use the `systemctl cat` and `systemctl show` commands
* recommending against modifying system (package) unit files in `/usr` and `/lib`, and exclusively using overrides

The second point goes against a sentence in the current docs ("Or, you may need to edit the docker.service file, which can be in /usr/lib/systemd/system, /etc/systemd/service, or /lib/systemd/system.") but I think this is definitely not recommended in the [systemd docs](http://www.freedesktop.org/software/systemd/man/systemd.unit.html#idm140118588064480).

I'll create a pull request to propose the change.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #13796.

### CODE REVIEW

**Comment by vdemeester** [comment]
:thumbsup:

I think it would be cool to add stuff about *unmasking* stuff too. Didn't test it (just helped on IRC), but there seem to be some distribution (ubuntu 15.04 server I think it was), where the ``docker.service`` and ``docker.socket`` were masked (https://botbot.me/freenode/docker/search/?q=unmask).

Something like "first check it's not masked, if it is unmask it, etc…".

@eolamey what do you think ?
/cc @thaJeztah @moxiegirl

File: docs/sources/articles/systemd.md [hunk 1]
```diff
@@ -24,14 +24,38 @@ If you want Docker to start at boot, you should also:
...
+    $ sudo systemctl show docker --property EnvironmentFile
+    EnvironmentFile=-/etc/sysconfig/docker
+
+Otherwise, you can see the location and content of the service files:
+
+    $ sudo systemctl cat docker
+    # /lib/systemd/system/docker.service
+    [Unit]
+    Description=Docker Application Container Engine
+    Documentation=http://docs.docker.com
```
Comment (thaJeztah) [inline docs/sources/articles/systemd.md]: Oh! Not your fault, but that doesn't look right, that should be `https://`. Looks like a mistake here; https://github.com/docker/docker/blob/9944f8b95453e3d0726fc037aaed41a98e72e070/contrib/init/systemd/docker.service#L3

Comment (eolamey) [inline docs/sources/articles/systemd.md]: You definitely have a very good eye, @thaJeztah :) I'll update the PR and fix the other link in the article.
I can submit a PR to fix the `docker.service` as well if you want.

Comment (eolamey) [inline docs/sources/articles/systemd.md]: Actually, there are a bunch of links to `http://docs.docker.com` in the docs, almost as much as to `https://docs.docker.com`. I can submit another PR to fix all of them, if it's OK with you.

File: docs/sources/articles/systemd.md [hunk 2]
```diff
@@ -24,14 +24,38 @@ If you want Docker to start at boot, you should also:
...
+    LimitNOFILE=1048576
+    LimitNPROC=1048576
+    LimitCORE=infinity
+
+    [Install]
+    WantedBy=multi-user.target
+
+The files located in `/usr/lib/systemd/system` or `/lib/systemd/system` should not
+be edited. Instead, you should override the properties you need to change by
+following [the second example](#http-proxy).
```
Comment (thaJeztah) [inline docs/sources/articles/systemd.md]: I'm thinking of moving the instructions for customizing first, and the 'warning' after that.
Also more explicitly refering to "which" example we're refering to.

Something like:

```Markdown
You can customize the Docker daemon options using override files as explained in the
[HTTP Proxy example](#http-proxy) below. The files located in `/usr/lib/systemd/system` 
or `/lib/systemd/system` contain the default options and should not be edited.
```

Perhaps @moxiegirl has a better suggestion

**Comment by eolamey** [comment]
@vdemeester I'm actually not that familiar with systemd, I learned about the `cat` and `show` command today while looking at the docs. I'm not sure I understand the systemd docs about `masking`/`unmasking`, or why someone would end up in a situation where they have to `unmask`. Doesn't that mean that the service has been explicitly `masked` before, by a user? I doubt that would be done by default by the package, would it?

I actually use Ubuntu 15.04, but installed the package from Docker, not Ubuntu. I'll check if there's any difference.

File: docs/sources/articles/systemd.md [hunk 3]
```diff
@@ -24,14 +24,38 @@ If you want Docker to start at boot, you should also:
 ## Custom Docker daemon options
 
 There are a number of ways to configure the daemon flags and environment variables
-for your Docker daemon. 
+for your Docker daemon.
 
 If the `docker.service` file is set to use an `EnvironmentFile`
 (often pointing to `/etc/sysconfig/docker`) then you can modify the
-referenced file.
+referenced file. To check if this is the case:
```
Comment (thaJeztah) [inline docs/sources/articles/systemd.md]: I suggest;

    Check if the `docker.service` uses an `EnvironmentFile`:

**Comment by thaJeztah** [comment]
Thanks for this, @eolamey. I added some suggestions, but possibly @moxiegirl has additional thoughts.

@vdemeester Perhaps we can add a 'troubleshooting' section?

**Comment by vdemeester** [comment]
@eolamey Ubuntu 15.04 version desktop or server ? I'll have to try.
@thaJeztah Yeah I think it would fit perfectly in a "Troubleshooting" section :wink:

File: docs/sources/articles/systemd.md (hunk 1 above)
Comment (vdemeester) [inline docs/sources/articles/systemd.md]: :+1:

Comment (thaJeztah) [inline docs/sources/articles/systemd.md]: that'd be great.

given the upcoming release, perhaps you can create a separate PR just for the unit file and "ping" me on that, as I think it should be cherry-picked in the release branch

**Comment by eolamey** [comment]
Thank @thaJeztah, I'll wait a bit to see if @moxiegirl has other suggestions, but otherwise I'll update with your comments. And I'll prepare PRs for the docs URL and the service file fix.

As for Ubuntu 15.04, I use neither Desktop nor Server, but a custom, minimalistic preseed. I don't think there's any difference when you install Ubuntu's `docker.io` package anyway.
I just tried it on a fresh VM, and didn't have to `unmask` the service but... now I see that the `show` command doesn't work on the `EnvironmentFile` property (it works on all others) -_-'

I'll update my PR to use `grep` instead of `--property` just for `EnvironmentFile`.

**Comment by vdemeester** [comment]
@eolamey Ok then a Troubleshooting section would be perfect.

**Comment by eolamey** [comment]
I am not really confortable writing a 'troubleshooting' section if I haven't experienced the problems myself. Would you mind contributing it, either here as a comment or in a separate PR?

**Comment by thaJeztah** [comment]
> I am not really confortable writing a 'troubleshooting' section if I haven't experienced the problems myself. Would you mind contributing it, either here as a comment or in a separate PR?

I think a separate PR would be good here. We may need to make a decision on that as well; If we're unable to reproduce the masking issue (and it only was a single incident), I'm not sure we should include it in the docs; we should guard against writing a full tutorial on systemd.

So a separate (follow-up) PR would be better, to prevent that discussion from delaying this PR.

**Comment by vdemeester** [comment]
@thaJeztah @eolamey Sounds ok with me :) Another PR would be better. For the "Troubleshooting", it depends on if we can reproduce or not, but also if it seems to hit a lot (based on the question on irc, stackoverflow, …) or not :smile_cat:.

**Comment by eolamey** [comment]
Good.

Now... for completeness, I installed a fresh CentOS 7 VM, and the service still isn't masked. The bad news is that the `systemctl cat` command is not available :( Should I just remove that part from the docs instead of warning it might not work?

**Comment by thaJeztah** [comment]
Quick google; looks like `systemctl cat` was added in [systemd version 209]( https://www.digitalocean.com/community/tutorials/how-to-use-systemctl-to-manage-systemd-services-and-units#displaying-a-unit-file) :disappointed:. Can we come up with a useful alternative?

**Comment by eolamey** [comment]
I looked into it but couldn't find any. I think the most important, for the docs, is being able to check if an `EnvironmentFile` is used, which works with `systemctl show` and `grep` for both Ubuntu and CentOS/RHEL. It may be best to remove `systemctl cat` altogether from the docs.

**Comment by eolamey** [comment]
We could show something like (so that the user can see where the service file is):

```shell
$ sudo systemctl status docker | grep Loaded
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled)
```

**Comment by thaJeztah** [comment]
> We could show something like (so that the user can see where the service file is)

Sounds good; I think that's the important part; finding where the file can be found

**Comment by eolamey** [comment]
All right, I updated the PR, and had to change the wording to replace `systemctl cat` with the `status` and `grep` commands.

**Comment by thaJeztah** [comment]
Thanks, @eolamey LGTM

ping @fredlf @jamtur01 @moxiegirl @SvenDowideit

**Comment by jamtur01** [comment]
LGTM - interesting.

**Comment by thaJeztah** [comment]
@moxiegirl should this be cherry-picked?

**Comment by moxiegirl** [comment]
@thaJeztah @jfrazelle yes please a cherry-pick --- Jess has been doing those into the RC

**Comment by jfrazelle** [comment]
cherry-picked

**Pull request merged**